T1000 e initialization

Dylanliacc opened the pull request:
Update V0.9.0

Dylanliacc committed:
feature-T1000-E: add Added the board definition for T1000-E

-  integrate a script for rapid dependency download that is compatible with both Linux and Windows platforms.
-  add the pin definitions for UART, SPI, GPIO, and other peripherals have been ensured to be correct.
-  add the env configuration for PlatformIO.

Dylanliacc committed:
Merge remote-tracking branch 'origin/master' into T1000-E_Initialization

Dylanliacc committed:
refact-T1000-E: redefine T1000-E board

Dylanliacc committed:
feature-T1000-E: add basic sensors

Dylanliacc committed:
feature-T1000-E: add button init

Dylanliacc committed:
Merge remote-tracking branch 'origin/master' into T1000-E_Initialization

Dylanliacc committed:
Merge branch 'master' into T1000-E_Initialization

Dylanliacc committed:
feat: add DRADIOLIB_GODMODE defination for use function setDioAsRfSwitch to DIO LORA RF

Dylanliacc committed:
feat : add gps(GNSS_Airoha) sleep mode

Dylanliacc committed:
feat: add behavier when rec or send message

Dylanliacc committed:
chore: hang IIC bus usage to avoid sensor address conflict

Dylanliacc committed:
feat: add sensor data acquisition

Dylanliacc committed:
feat : support Airoha GPS

- add disable it in FSM
- update  lookForTime and lookForLocation function

Dylanliacc committed:
fix: fix a bug

Dylanliacc committed:
Merge branch 'master' into T1000-E_Initialization

Dylanliacc committed:
version: change version to 0.9.0

fifieldt commented on variants/tracker-t1000-e/platformio.ini:
```diff
@@ -0,0 +1,15 @@
+board = tracker-t1000-e
+board_level = extra
+; platform = https://github.com/maxgerhardt/platform-nordicnrf52#cac6fcf943a41accd2aeb4f3659ae297a73f422e
+build_flags = ${nrf52840_base.build_flags} -Ivariants/tracker-t1000-e -DTRACKER_T1000_E  -DRADIOLIB_GODMODE
```
覺得人會抗議GODMODE. 有可能需要在RadioLib Upstream改改

Dylanliacc replied:
For now, let's do it this way. If anyone has any objections, we can make changes later~

fifieldt commented on variants/tracker-t1000-e/run_once.sh:
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
```
這檔案沒錯誤，但可能不需要加。在meshtastic使用bin/build-nrf52.sh (已經有mergehex和softdevice).

Dylanliacc replied:
done deleted

fifieldt commented on variants/tracker-t1000-e/variant.h:
```diff
@@ -0,0 +1,158 @@
+#define AREF_VOLTAGE 3.0
+#define VBAT_AR_INTERNAL AR_INTERNAL_3_0
+
+// #define ADC_CTRL (32 + 6)
```
consider removing comment before submitting patch

Dylanliacc replied:
removed that

fifieldt commented on variants/tracker-t1000-e/variant.h:
```diff
@@ -0,0 +1,158 @@
+
+#define ADC_RESOLUTION 14
+#define BATTERY_SENSE_RESOLUTION_BITS 12
+// #define BATTERY_SENSE_RESOLUTION 4096.0
```
consider removing comment before submitting patch

Dylanliacc replied:
removed that

fifieldt commented on variants/tracker-t1000-e/variant.h:
```diff
@@ -0,0 +1,158 @@
+#define GPS_FIX_HOLD_TIME 15000 // ms
+
+#define BATTERY_PIN 2
+// #define ADC_CHANNEL ADC1_GPIO2_CHANNEL
```
consider removing comment before submitting patch

Dylanliacc replied:
done for that

fifieldt commented on variants/tracker-t1000-e/variant.h:
```diff
@@ -0,0 +1,158 @@
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef _VARIANT_WIO_SDK_WM1110_
```
is a new VARIANT name needed?

Dylanliacc replied:
changed it to _VARIANT_TRACKER_T1000_E_

fifieldt commented on boards/tracker-t1000-e.json:
```diff
@@ -0,0 +1,58 @@
+{
```
Looks good.

fifieldt commented on variants/tracker-t1000-e/platformio.ini:
```diff
@@ -0,0 +1,15 @@
+board = tracker-t1000-e
+board_level = extra
+; platform = https://github.com/maxgerhardt/platform-nordicnrf52#cac6fcf943a41accd2aeb4f3659ae297a73f422e
+build_flags = ${nrf52840_base.build_flags} -Ivariants/tracker-t1000-e -DTRACKER_T1000_E  -DRADIOLIB_GODMODE
```
Does this need  -Isrc/platform/nrf52/softdevice -Isrc/platform/nrf52/softdevice/nrf52 ?

Dylanliacc replied:
changed it to the softdevice now

fifieldt commented on variants/tracker-t1000-e/variant.cpp:
```diff
@@ -0,0 +1,64 @@
+/*
```
Looks good

fifieldt commented on src/modules/Telemetry/Sensor/T1000xSensor.cpp:
```diff
@@ -0,0 +1,128 @@
+#include "configuration.h"
```
ok!

fifieldt commented on src/modules/Telemetry/Sensor/T1000xSensor.h:
```diff
@@ -0,0 +1,24 @@
+#include "configuration.h"
```
ok!

fifieldt commented on src/modules/TextMessageModule.cpp:
```diff
@@ -12,7 +12,11 @@ ProcessMessage TextMessageModule::handleReceived(const meshtastic_MeshPacket &mp
     LOG_INFO("Received text msg from=0x%0x, id=0x%x, msg=%.*s\n", mp.from, mp.id, p.payload.size, p.payload.bytes);
 #endif
-
+#ifdef PIN_BUZZER 
```
This turns on the buzzer for all variants. Is this what the community wants?

fifieldt commented on src/gps/GPS.cpp:
```diff
@@ -400,6 +400,12 @@ bool GPS::setup()
     int msglen = 0;
 
     if (!didSerialInit) {
+#ifdef GNSS_Airoha 
```
See https://github.com/fifieldt/meshtastic-firmware/commit/7b6b960671e1e2787003bd903019e20f8be1cf5b

markbirss committed:
Update tracker-t1000-e.json

Remove a space

markbirss committed:
Delete variants/tracker-t1000-e/run_once.sh

Delete not need as we will change platformio.ini

markbirss committed:
Update platformio.ini

Update SoftDevice 7.3.0 usage in line with other lr1110 targets

Do we need to keep GODMODE ?

Dylanliacc committed:
Merge remote-tracking branch 'origin/markbirss-patch-2' into T1000-E_Initialization

Dylanliacc committed:
Merge remote-tracking branch 'origin/markbirss-patch-3' into T1000-E_Initialization

Dylanliacc committed:
Merge branch 'master' into T1000-E_Initialization
